packages/components/react-web/src/Input/Input.jsx

```diff
@@ -16,18 +14,28 @@ import { inputStyles } from './Input.styles';
  * For accessibility purposes, every instance of a form element must be
  * accompanied by a label, even if it’s visually hidden in the interface.
  */
-export const Input = ({ type, size, label, disabled, validator, ...props }) => {
+export const Input = ({
+  id,
+  type,
+  size,
+  label,
+  disabled,
+  validator,
+  value: defaultValue,
+  onChange,
+  ...props
+}) => {
   // State used for text input fields
-  const [value, setValue] = useState('');
+  const [value, setValue] = useState(defaultValue);
   const [disable, setDisable] = useState(disabled);
   const [touched, setTouched] = useState(false);
   const [errorMessage, setErrorMessage] = useState('');
 
   // State used for radio buttons and checkboxes
   const [checked, setChecked] = useState(false);
 
-  const [inputId] = useState(helpers.gdsId());
-  const [errorId] = useState(helpers.gdsId());
+  const inputId = id ?? helpers.gdsId();
+  const errorId = helpers.gdsId();
```

Comment:
`helpers.gdsId()` will set a different id each time the component re-renders
I think the old code used `useState` so a value can be captured only once